### PR TITLE
`TypeError` if `_initialize_models` called with locales set on table

### LINKED ISSUE #2232: `TypeError` if `_initialize_models` called with locales set on table
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version:
* Python version:
* Operating System:

### Error Description
Calling `BaseMultiTableSynthesizer._initialize_models` with locales set for any table causes `TypeError: sdv.single_table.copulas.GaussianCopulaSynthesizer() got multiple values for keyword argument 'locales'`

### Steps to reproduce

```python
from sdv.datasets.demo import download_demo
from sdv.multi_table import HMASynthesizer

data, metadata = download_demo('multi_table', 'fake_hotels')
synthesizer = HMASynthesizer(metadata)
synthesizer.set_table_parameters('guests', {'locales': ['en_CA']})
synthesizer._initialize_models()
```

### PR DESCRIPTION
**Pull request by frances-h**
Resolve #2232 
CU-86b27kh49

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b27kh49 SDV - `TypeError` if `_initialize_models` called with locales set on table #2232](https://app.clickup.com/t/86b27kh49)

File: tests/unit/multi_table/test_base.py [hunk 1]
```diff
@@ -40,7 +40,9 @@ def test__initialize_models(self):
         locales = ['en_CA', 'fr_CA']
         instance = Mock()
         instance._table_synthesizers = {}
-        instance._table_parameters = {'nesreca': {'default_distribution': 'gamma'}}
+        instance._table_parameters = {
+            'nesreca': {'default_distribution': 'gamma', 'locales': ['fr_CA', 'en_CA']},
```
Comment (gsheni) [inline tests/unit/multi_table/test_base.py:44]: Use previous variable:
```suggestion
            'nesreca': {'default_distribution': 'gamma', 'locales': locales[::-1]},
```

File: tests/unit/multi_table/test_base.py [hunk 2]
```diff
@@ -57,7 +59,7 @@ def test__initialize_models(self):
             call(
                 metadata=instance.metadata.tables['nesreca'],
                 default_distribution='gamma',
-                locales=locales,
+                locales=['fr_CA', 'en_CA'],
```
Comment (gsheni) [inline tests/unit/multi_table/test_base.py:62]:
```suggestion
                locales=locales[::-1],
```

File: tests/unit/multi_table/test_base.py (hunk 1 above)
Comment (frances-h) [reply tests/unit/multi_table/test_base.py]: For completeness I'll just use a completely different locale list 😅

**Review by gsheni — APPROVED** [review]

**Review by R-Palazzo — APPROVED** [review]

**Pull request merged**